src/Shared/UnitTests/MockEngine.cs

```diff
@@ -439,31 +433,25 @@ internal void AssertLogContains(string contains)
         /// First check if the string is in the log string. If not
         /// than make sure it is also not in the MockLogger
         /// </summary>
-        /// <param name="contains"></param>
         internal void AssertLogDoesntContain(string contains)
         {
-            if (_output == null)
+            var logText = _log.ToString();
+            if (logText.IndexOf(contains, StringComparison.OrdinalIgnoreCase) != -1)
             {
-                Console.WriteLine(_log);
-            }
+                if (_output == null)
+                {
+                    Console.WriteLine(logText);
+                }
+                else
+                {
+                    _output.WriteLine(logText);
+                }
 
-            if (_upperLog == null)
-            {
-                _upperLog = _log.ToString().ToUpperInvariant();
+                // If we do not contain this string than pass it to
+                // MockLogger. Since MockLogger is also registered as
+                // a logger it may have this string.
+                _mockLogger.AssertLogDoesntContain(contains);
```

Comment:
Shouldn't there be an assert in here somewhere? If the log string doesn't contain the message, that part passes and it should check _mockLogger. If it does contain the message, it should print and fail without bothering.  Right?